File: lab9/lab25-1/Bibliotecas/Vehiculo.cpp
//
// Created by ANTONIO on 25/06/2026.
//

#include "Vehiculo.h"

Vehiculo::Vehiculo() = default;

Vehiculo::Vehiculo(const Vehiculo &origin) {
    *this = origin;
}

string Vehiculo::get_placa() const {
    return placa;
}
void Vehiculo::set_placa(const string &placa) {
    this->placa = placa;
}

string Vehiculo::get_marca() const {
    return marca;
}
void Vehiculo::set_marca(const string &marca) {
    this->marca = marca;
}

string Vehiculo::get_modelo() const {
    return modelo;
}
void Vehiculo::set_modelo(const string &modelo) {
    this->modelo = modelo;
}
//
ifstream& Vehiculo::leer(ifstream &arch) {
    string placaVehiculo,marcaVehiculo,modelVehiculo;

    getline(arch,placaVehiculo,',');
    getline(arch,marcaVehiculo,',');
    getline(arch,modelVehiculo);

    this->set_placa(placaVehiculo);
    this->set_marca(marcaVehiculo);
    this->set_modelo(modelVehiculo);

    return arch;
}
ofstream& Vehiculo::imprimir(ofstream &archRep) const {
    archRep<<this->get_placa()<<" "<<this->get_marca()<<" "<<this->get_modelo()<<endl;
    return archRep;
}
//
Vehiculo& Vehiculo::operator= (const Vehiculo &origin) {
    if (this!=&origin) {
        this->set_placa(origin.get_placa());
        this->set_marca(origin.get_marca());
        this->set_modelo(origin.get_modelo());
    }
    return *this;
}

//destructor
Vehiculo::~Vehiculo() = default;
//
ifstream& operator>> (ifstream &arch, Vehiculo &vehiculo) {
    return vehiculo.leer(arch);
}
ofstream& operator<< (ofstream &archRep, const Vehiculo &vehiculo) {
    return vehiculo.imprimir(archRep);
}

File: lab10/lab25-1/src/Vehiculo.java
import java.util.Scanner;

public class Vehiculo extends Registro {
    private String placa;
    private String marca;
    private String modelo;
    private int anioFab;
    private String categoria;
    private Propietario propietario;

    public Vehiculo() {
    }

    public Vehiculo(Vehiculo other) {
        this.placa = other.placa;
        this.marca = other.marca;
        this.modelo = other.modelo;
        this.anioFab = other.anioFab;
        this.categoria = other.categoria;
        this.propietario = new Propietario(other.propietario);
    }

    public String getPlaca() {
        return placa;
    }

    public void setPlaca(String placa) {
        this.placa = placa;
    }

    public String getMarca() {
        return marca;
    }

    public void setMarca(String marca) {
        this.marca = marca;
    }

    public String getModelo() {
        return modelo;
    }

    public void setModelo(String modelo) {
        this.modelo = modelo;
    }

    public int getAnioFab() {
        return anioFab;
    }

    public void setAnioFab(int anioFab) {
        this.anioFab = anioFab;
    }

    public String getCategoria() {
        return categoria;
    }

    public void setCategoria(String categoria) {
        this.categoria = categoria;
    }

    public Propietario getPropietario() {
        return propietario;
    }

    public void setPropietario(Propietario propietario) {
        this.propietario = propietario;
    }

    @Override
    public void cargar(Scanner archivo){
        if(!archivo.hasNext()){
            return;
        }
        this.placa = archivo.next();
        this.marca = archivo.next();
        this.modelo = archivo.next();
        this.anioFab = archivo.nextInt();
        this.categoria = archivo.next();
        this.propietario.setDni(archivo.nextInt());

    }
    @Override
    public void imprimir(){
        System.out.println(this);
    }
    @Override
    public String toString() {
        return String.format(
                "Placa: %s\n Marca: %s\n Modelo: %s\n Año de fabricacion: %d\n Categoria: %s\n",
                placa,
                marca,
                modelo,
                anioFab,
                categoria
        );
    }

}
